File: 【视频】03 MPMoviePlayerViewController播放视频/OC-MPMoviePlayerViewController播放视频/OC-MPMoviePlayerViewController播放视频/ViewController.m
//
//  ViewController.m
//  OC-MPMoviePlayerViewController播放视频
//
//  Created by 蓝田 on 2016/11/9.
//  Copyright © 2016年 蓝田. All rights reserved.
//


#import "ViewController.h"
#import <MediaPlayer/MediaPlayer.h>

@interface ViewController ()
@property (nonatomic, strong) MPMoviePlayerViewController *playerVC;
@end

@implementation ViewController

-(MPMoviePlayerViewController *)playerVC
{
    if (!_playerVC) {
        // ① 本地媒体文件
        // 方法1：
        NSString *path = [[NSBundle mainBundle]pathForResource:@"1.mp4" ofType:nil];
        NSURL *url = [NSURL fileURLWithPath:path];
        
        // 方法2：
        // NSURL *url = [[NSBundle mainBundle]URLForResource:@"1.mp4" withExtension:nil];
        
        _playerVC = [[MPMoviePlayerViewController alloc] initWithContentURL:url];
        
        
        // ② 通过远程URL, 创建控制器 MPMoviePlayerViewController
        // NSURL *remoteURL = [NSURL URLWithString:@"http://v1.mukewang.com/57de8272-38a2-4cae-b734-ac55ab528aa8/L.mp4"];
        // _playerVC = [[MPMoviePlayerViewController alloc] initWithContentURL:remoteURL];
    }
    return _playerVC;
}


- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event{
    
    // 2. 直接模态弹出该控制器(或者: 设置播放视图frame, 添加到需要展示的视图上)
    [self presentViewController:self.playerVC
                       animated:YES completion:^{
                           // 3. 开始播放
                           [self.playerVC.moviePlayer play];
                       }];
}

@end

File: 【视频】03 MPMoviePlayerViewController播放视频/Swift-MPMoviePlayerViewController播放视频/Swift-MPMoviePlayerViewController播放视频/ViewController.swift
//
//  ViewController.swift
//  Swift-MPMoviePlayerViewController播放视频
//
//  Created by 蓝田 on 2016/11/9.
//  Copyright © 2016年 蓝田. All rights reserved.
//

import UIKit
import MediaPlayer

class ViewController: UIViewController {
    
    lazy var playerVC: MPMoviePlayerViewController = {
        
        // ① 本地媒体文件
        let infoPlistPath = Bundle.main.url(forResource: "1", withExtension: "mp4")
        let vc = MPMoviePlayerViewController(contentURL: infoPlistPath!)
        
        // ② 通过远程URL, 创建控制器 MPMoviePlayerViewController
        // let url = URL(string: "http://v1.mukewang.com/a45016f4-08d6-4277-abe6-bcfd5244c201/L.mp4")
        // let vc = MPMoviePlayerViewController(contentURL: url!)
        
        return vc!
    }()
    
    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        
        // 表示已经弹出控制器，则直接return
        if presentedViewController != nil {
            return
        }
        
        // 2. 直接模态弹出该控制器(或者: 设置播放视图frame, 添加到需要展示的视图上)
        present(playerVC, animated: true) {
            // 3. 开始播放
            self.playerVC.moviePlayer.play()
        }
    }
}
